src/vm/appdomain.cpp

```diff
@@ -6368,6 +6369,8 @@ HRESULT RuntimeInvokeHostAssemblyResolver(INT_PTR pManagedAssemblyLoadContextToB
         {
             // Step 2 (of CLRPrivBinderAssemblyLoadContext::BindUsingAssemblyName) - Invoke Load method
             // This is not invoked for TPA Binder since it always returns NULL.
+            tracer.Trace(BinderTracing::ResolutionAttemptedOperation::Stage::FindInLoadContext,
```

Comment:
Should this call to `Trace` be for the stage (calling `AssemblyLoadContext.Load`) that is about to start?
`FindInLoadContext` -> `ALCLoad`

Reply:
In general I think it's more valuable to have traces after each of the stages including the result of the stage. For example in this case of calling `Load` it is very interesting to know if it returns `null` or something else (and what). As that is easy to map back to the code.
Having an event that we're about to call `Load`, then followed by another even that we're about to fallback to TPA - I can deduce what happened I guess, but it makes it much harder to read. And having the event before `Load` feels like not adding any actual value on its own.

Reply:
Yeah, @lpereira and I chatted about how the result of each stage is what we really need / care about. He's planning on updating to that.

Reply:
Yes, the incoming change will basically generate two events for each stage, one when it starts, and one when it finishes.  The rationale is that this can be used for timing each step, but if two events for each stage is too much, this behavior can be modified later.  The new ResolutionAttemptedOperation class will fire events with whatever context it's provided with at the time it goes to a different stage.